src/Background/NotificationHandler.vala

```diff
@@ -0,0 +1,109 @@
+public class NotificationHandler : Object {
+    public const string ACTION_ALLOW_BACKGROUND = "background.allow";
+    public const string ACTION_FORBID_BACKGROUND = "background.forbid";
+
+    private DBusConnection connection;
+    private HashTable<uint32, NotificationRequest> notification_by_id;
+    private Notifications notifications;
+
+    private enum NotifyBackgroundResult {
+        FORBID,
+        ALLOW,
+        ALLOW_ONCE
+    }
```

Comment:
Regarding the FAILED entry: I think it might be better to just respond with ALLOW_ONCE because if we fail to send a notification IMHO it's best to just let the app silently continue to run in the background.